test/unit/transactor/status_test.rb

```diff
@@ -237,5 +237,33 @@ def setup
       assert_nil           month['exceeded']
       assert_equal 'true', day['exceeded']
     end
+
+    test '#to_xml shows the application id when OAuth is used' do
+      usage  = {:month => {@metric_id.to_s => 429}}
+      status = Transactor::Status.new(service_id: @service_id,
+                                      application: @application,
+                                      values: usage,
+                                      oauth: true)
+
+      doc = Nokogiri::XML(status.to_xml)
+
+      assert_equal @application.id,  doc.at('status application id').content
+    end
+
+    test '#to_xml shows the username along with the application when OAuth and a User are used' do
+      usage  = {:month => {@metric_id.to_s => 429}}
+      user = User.new(service_id: @service_id,
+                       username: "testusername")
+      status = Transactor::Status.new(service_id: @service_id,
+                                      application: @application,
+                                      values: usage,
+                                      oauth: true,
+                                      user: user
+                                      )
+      doc = Nokogiri::XML(status.to_xml)
+
+      assert_equal "testusername", doc.at('status user id').content
```

Comment:
You could define `test_user_name` at the beginning of the test and then use it to when instantiating the user and also here. It would avoid duplication.

Reply:
+1